pandas/plotting/_core.py

```diff
@@ -815,11 +816,22 @@ def _post_plot_logic(self, ax, data):
 class ScatterPlot(PlanePlot):
     _kind = 'scatter'
 
-    def __init__(self, data, x, y, s=None, c=None, **kwargs):
+    def __init__(self, data, x, y, s=None, c=None, size_factor=1, **kwargs):
         if s is None:
-            # hide the matplotlib default for size, in case we want to change
-            # the handling of this argument later
+            # Set default size if no argument is given.
             s = 20
+        elif is_hashable(s) and s in data.columns:
+            # Handle the case where s is a label of a column of the df.
+            # The data is normalized to 200 * size_factor.
+            size_data = data.loc[:, s].values
+            if is_numeric_dtype(size_data):
+                self.size_title = s
+                self.s_data_max = size_data.max()
+                self.size_factor = size_factor
+                self.bubble_points = 200
+                s = self.bubble_points * size_factor * size_data / self.s_data_max
+            else:
```

Comment:
I'd like to support Categorical data here as well. How much work would it be to do that? Perhaps

```python
if is_categorical_dtype(size_data):
    if size_data.ordered:
        size_data = size_data.codes
    # else raise with a nice error message
```

and then the `if is_numeric_dtype(size_data)`? Does that work?

Reply:
@TomAugspurger I'm not sure I understand what it means to plot categorical data as sizes. Could you give me a use case example?

Reply:
@VincentAntoine this basically, but with all the nice stuff from this PR

```python
df = pd.DataFrame(np.random.randn(100, 2), columns=['a', 'b'])

categories = list('abcd')
df['c'] = pd.Categorical(np.random.choice(categories, size=(100,)),
                         categories=categories, ordered=True)
df.head()

fig, ax = plt.subplots()
s = (10 + df.c.cat.codes * 10)

ax.scatter(x='a', y='b', data=df, s=s);
```


![gh](https://user-images.githubusercontent.com/1312546/31178441-b646a4e8-a8de-11e7-9af6-1feacba75172.png)


So the idea is to automatically know to use `.categories.codes` for categorical dtype data, instead of just the categories (which may not be numeric). I think if you do the

- check for categorical dtype
- size_data = `df[s].cat.codes`


before `if is_numeric_dtype(size_data)`, then everything else should be the same

Reply:
Yes, it works just as you said with no additionnal modification :)

Reply:
We just need to do size_data = df[s].cat.codes + 1, as codes start at 0 and the resulting bubbles will have an area of 0.

Reply:
@TomAugspurger I added this in the last commit, and added a test for scatter plot with categorical data as well. I'll write the release note now. Let me know if the code needs any more changes.